internal/pkg/daemon/bpfrecorder/bpfrecorder.go

```diff
@@ -546,16 +546,17 @@ func toStringByte(array []byte) string {
 func (b *BpfRecorder) processEvents(events chan []byte) {
 	b.logger.Info("Processing events")
 	for event := range events {
-		func() {
+		eventCopy := event
+		go func() {
```

Comment:
While this is a good enhancement, I'm slightly concerned that we'd hit a goroutine limit in bigger deployments. Pheraps after this merges it would be good to do some stress testing of this part of the code-base.

Reply:
In theory we can get 4194304 go routines for 64bit systems (max amount of pids), which could cause memory spikes. So yeah either we limit it by using a worker pool or we adjust the memory constraints accordingly.

Reply:
Would it help, as an additional enhancement, to have a pool of gouroutines? I'm sure we're not the first ones with this requirement, so I bet there exists a library that already does this..

Reply:
I think @saschagrunert 's solution to use a semaphore is equivalent to having a pool. I don't have an opinion in terms of making that configurable as @pjbgf suggested, but it might be worth doing.